### ISSUE TITLE
API Auth V2 SUPPORT

### ISSUE THREAD

**Issue by thaitanloi365** [teacat/shopeego#1]
Hi there !

Do you have a plan to support API Auth V2? 

Thanks you for your great works.

**Comment by YamiOdymel** [comment]
Do you have any information for V2? For example, does all V1 APIs supports V2 if I just changed the URL from `/v1` to `/v2` with the new V2 auth params?

The documentation of Shopee that official provided was kinda retarded, I'm wondering if you have any information could help. I might try to support V2 but maybe not an emergency because V1 still does work.

https://open.shopee.com/documents?module=63&type=2&id=56

**Comment by thaitanloi365** [comment]
Hi @YamiOdymel 

Actually, apis v1 are working correctly. But in my application, I need to refresh acccess_token when it was expired, right now I direct the customer to oauth url to re-auth. But in v2 they provide a api to refresh_token, so I think it should be better. Do you have any ideals about that situation ? 

Thanks you.

**Comment by YamiOdymel** [comment]
Huh, spent a while with V2 and it's really- retarded (_thanks to the Shopee documentation_), but I'm implementing. Might be finished this weekend.

**Comment by YamiOdymel** [comment]
Well, bad news. I implemented the V2 features then I realized there were **NO** available V2 API could be used. 

Shopee offers you Open API V2 but the API were **ALL** V1 (excepted the `GetAccessToken `and `RefreshAccessToken` are actually V2 and can be called). I changed the URLs from `/v1` to `/v2` and it all goes `404 not found`.

I don't know when did the Open API V2 appear, but it seems not fully-implemented. I suggest to implement the Token System by yourself, it's way better than this half-broken- refresh token thing. 

Atleast I've wrote down the V2 base in the package though.
